h-include.js

```diff
@@ -41,35 +41,64 @@ var hinclude;
   hinclude = {
     classprefix: "include_",
 
-    show_content: function (element, req) {
-      var i, include, message, fragment = element.getAttribute('fragment') || 'body';
-      if (req.status === 200 || req.status === 304) {
-        var container = document.implementation.createHTMLDocument().documentElement;
-        container.innerHTML = req.responseText;
+    create_container: function (req) {
+      var container = document.implementation.createHTMLDocument().documentElement;
+      container.innerHTML = req.responseText;
+
+      return container;
+    },
+    check_recursion: function (container, element) {
```

Comment:
Actually `check_recursion()` doesn't use `container` so it can be called as is from `include()`. 
That's how you ended up implementing the check in #19.